runtime/service.go

```diff
@@ -1278,8 +1278,14 @@ func (s *service) Cleanup(requestCtx context.Context) (*taskAPI.DeleteResponse,
 }
 
 func (s *service) monitorVMExit() {
-	// once the VM shuts down, the shim should too
-	defer s.shimCancel()
+	defer func() {
+		// we ignore the error here due to cleanup will only succeed if the jailing
+		// process was killed via SIGKILL
+		s.jailer.Close()
```

Comment:
So if the jailing process (firecracker) was not killed by SIGKILL, would `Close()` return an error? Can we specifically check the known, expected error and log other errors?

Reply:
Yea, it would return a generic "exit status 1" error. Which isn't too useful. We can do a string compare but it'd be pretty hacky. My idea would be to capture the `Stderr` from the command and do a string compare and return a better error instead.

What do you think?

Reply:
How about using `runc delete -f`?

Reply:
I've went ahead and added a `errContainerDoesNotExist` error to check against.